vm/class_test.go

```diff
@@ -453,4 +462,123 @@ func TestRequireFail(t *testing.T) {
 	if err.Message != expected {
 		t.Fatalf("Error message should be '%s'. got: %s", expected, err.Message)
 	}
+	vm.checkCFP(t, 0, 1)
+}
+
+func TestGeneralIsNilMethod(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected bool
+	}{
+		{`123.is_nil`, false},
+		{`"Hello World".is_nil`, false},
+		{`(2..10).is_nil`, false},
+		{`{ a: 1, b: "2", c: ["Goby", 123] }.is_nil`, false},
+		{`[1, 2, 3, 4, 5].is_nil`, false},
+		{`true.is_nil`, false},
+		{`String.is_nil`, false},
+		{`nil.is_nil`, true},
+	}
+
+	for i, tt := range tests {
+		vm := initTestVM()
+		evaluated := vm.testEval(t, tt.input)
+		checkExpected(t, i, evaluated, tt.expected)
+		vm.checkCFP(t, i, 0)
+	}
+}
+
+func TestGeneralIsNilMethodFail(t *testing.T) {
+	testsFail := []struct {
+		input  string
+		errMsg string
+	}{
+		{`123.is_nil("Hello")`, "ArgumentError: Expect 0 argument. got=1"},
+		{`123.is_nil("Hello", "World")`, "ArgumentError: Expect 0 argument. got=2"},
+		{`"Fail".is_nil("Hello")`, "ArgumentError: Expect 0 argument. got=1"},
+		{`"Fail".is_nil("Hello", "World")`, "ArgumentError: Expect 0 argument. got=2"},
+		{`[1, 2, 3].is_nil("Hello")`, "ArgumentError: Expect 0 argument. got=1"},
+		{`[1, 2, 3].is_nil("Hello", "World")`, "ArgumentError: Expect 0 argument. got=2"},
+		{`{ a: 1, b: 2, c: 3 }.is_nil("Hello")`, "ArgumentError: Expect 0 argument. got=1"},
+		{`{ a: 1, b: 2, c: 3 }.is_nil("Hello", "World")`, "ArgumentError: Expect 0 argument. got=2"},
+		{`(1..10).is_nil("Hello")`, "ArgumentError: Expect 0 argument. got=1"},
+		{`(1..10).is_nil("Hello", "World")`, "ArgumentError: Expect 0 argument. got=2"},
+	}
+
+	for i, tt := range testsFail {
+		vm := initTestVM()
+		evaluated := vm.testEval(t, tt.input)
+		checkError(t, i, evaluated, ArgumentError, tt.errMsg)
+		vm.checkCFP(t, i, 1)
+	}
+}
+
+func TestGeneralIsAMethod(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected bool
+	}{
+		{`123.is_a(Integer)`, true},
+		{`123.is_a(Object)`, true},
+		{`123.is_a(String)`, false},
+		{`123.is_a(Range)`, false},
+		{`"Hello World".is_a(String)`, true},
+		{`"Hello World".is_a(Object)`, true},
+		{`"Hello World".is_a(Boolean)`, false},
+		{`"Hello World".is_a(Array)`, false},
+		{`(2..10).is_a(Range)`, true},
+		{`(2..10).is_a(Object)`, true},
+		{`(2..10).is_a(Null)`, false},
+		{`(2..10).is_a(Hash)`, false},
+		{`{ a: 1, b: "2", c: ["Goby", 123] }.is_a(Hash)`, true},
+		{`{ a: 1, b: "2", c: ["Goby", 123] }.is_a(Object)`, true},
+		{`{ a: 1, b: "2", c: ["Goby", 123] }.is_a(Class)`, false},
+		{`{ a: 1, b: "2", c: ["Goby", 123] }.is_a(Array)`, false},
+		{`[1, 2, 3, 4, 5].is_a(Array)`, true},
+		{`[1, 2, 3, 4, 5].is_a(Object)`, true},
+		{`[1, 2, 3, 4, 5].is_a(Null)`, false},
+		{`[1, 2, 3, 4, 5].is_a(String)`, false},
+		{`true.is_a(Boolean)`, true},
+		{`true.is_a(Object)`, true},
+		{`true.is_a(Array)`, false},
+		{`true.is_a(Integer)`, false},
+		// TODO: Fix Class.is_a Error when argument is a
+		//{`String.is_a(Class)`, true},
```

Comment:
In this test case, I try it in REPL and got the result:

```
$ go run goby.go -i
Goby 0.0.9
» String.is_a
ArgumentError: Expect 1 argument. got=0
#=> ERROR: ArgumentError: Expect 1 argument. got=0
» String.is_a(Class)
Internal Error: runtime error: index out of range
panic: runtime error: index out of range [recovered]
        panic: runtime error: index out of range

goroutine 1 [running]:
github.com/goby-lang/goby/vm.(*thread).execInstruction.func1(0xc42011b860)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/thread.go:71 +0x81
panic(0x1336140, 0x15392a0)
        /usr/local/Cellar/go/1.8.3/libexec/src/runtime/panic.go:489 +0x2cf
github.com/goby-lang/goby/vm.glob..func21(0xc42011b860, 0xc4201091f0, 0xc42011f400, 0x2, 0x2)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/instruction.go:343 +0x452
github.com/goby-lang/goby/vm.(*thread).execInstruction(0xc42011b860, 0xc4201091f0, 0xc420128fc0)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/thread.go:75 +0x96
github.com/goby-lang/goby/vm.(*thread).evalCallFrame(0xc42011b860, 0xc4201091f0)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/thread.go:42 +0x60
github.com/goby-lang/goby/vm.(*thread).startFromTopFrame(0xc42011b860)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/thread.go:36 +0x55
github.com/goby-lang/goby/vm.(*VM).startFromTopFrame(0xc420102240)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/vm.go:190 +0x2f
github.com/goby-lang/goby/vm.(*VM).REPLExec(0xc420102240, 0xc42011ceb0, 0x2, 0x2)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/repl.go:42 +0x64e
github.com/goby-lang/goby/igb.StartIgb(0x139a4df, 0x5)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/igb/repl.go:212 +0xfed
main.main()
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/goby.go:27 +0x657
exit status 2
```

I need some time to check out for the bug